src/tests/math/test_math_all_close.f90

```diff
@@ -0,0 +1,38 @@
+program tester
+
+    use stdlib_math, only: all_close
+    use stdlib_error, only: check
+    implicit none
+    
+    call test_math_all_close_real
+    call test_math_all_close_complex
+    print *, "All tests in `test_math_all_close` passed."
+
+contains
+
+    subroutine test_math_all_close_real
+
+        real :: x(4, 4), random(4, 4)
+
+        call random_number(random)
+        x = 1.0
+
+        call check(all_close(x+1.0e-11*random, x), msg="REAL: all_close(x+1.0e-11*random, x) failed.")
+        call check(all_close(x+1.0e-5 *random, x), msg="REAL: all_close(x+1.0e-5 *random, x) failed.", warn=.true.)
```

Comment:
Why do we always warn here?

Reply:
This may be because I thought it was more interesting when I wrote it **before**. I think I should be more rigorous. I will cancel the pseudo-random numbers here and replace them with exact numbers for comparison.
And I will consider the judgment of adding NAN. (I will set this PR as **a draft** until I improve these)
Thank you for your help :) @awvwgk

Reply:
I re-corrected the test program, this problem has been resolved now.